src/core/main.lua

```diff
@@ -33,39 +33,34 @@ function main ()
    zone("startup")
    require "lib.lua.strict"
    initialize()
-   local args = parse_command_line()
-   local program = programname(args[1])
-   if program == 'snabb' then
-      -- Print usage with exit status 0 if help requested
-      if args[2] == '-h' or args[2] == '--help' then
-         usage(0)
-      end
-      -- Print usage with exit status 1 if no arguments supplied
-      if #args == 1 then
-         usage(1)
-      end
-      -- Strip 'snabb' and use next argument as program name
-      table.remove(args, 1)
-   end
-   program = select_program(program, args)
+   local program, args = select_program(parse_command_line())
    if not lib.have_module(modulename(program)) then
       print("unsupported program: "..program:gsub("_", "-"))
-      print()
-      print("Rename this executable (cp, mv, ln) to choose a supported program:")
-      print("  snabb "..(require("programs_inc"):gsub("\n", " ")))
-      os.exit(1)
+      usage(1)
    else
       require(modulename(program)).run(args)
    end
 end
 
--- If program stars with prefix 'snabb_' removes the prefix
--- If not, use the next argument as program name
-function select_program (program, args)
-   if program:match("^snabb_") then
-      return program:gsub("^snabb_", "")
+-- Take the program name from the first argument, unless the first
+-- argument is "snabb", in which case pop it off, handle any options
+-- passed to snabb itself, and use the next argument.
+function select_program (args)
+   local program = programname(table.remove(args, 1))
+   if program == 'snabb' then
+      while #args > 0 and args[1]:match('^-') do
```

Comment:
@eugeneia I think there are two different hats that one can wear when reviewing code:

Community member: just giving feedback, nothing is binding, no responses are required.

Upstream maintainer: sole person responsible for accepting/merging the branch or spelling out explicitly what needs to be done before that can happen.

If you are the person who is resposible for upstreaming the PR and you don't press "Merge" then this is implicitly strong feedback "this code is not currently acceptable to me and so I have declined to merge it." In that case the contributor needs to clearly understand what are the deal-breakers so that they can fix them and the branch can be merged. The maintainer might say "You have to change A to B and remove C" or "we need to get an ack from @somebody or @someonelse" and then once that is done the contributor will expect the branch to merge.

This is why I have been stressing the importance of unambiguously dispatching incoming PRs between upstream maintainers. Then everybody knows who is wearing which hat and what is the path forward.

Overall there will be agreement between O(log(numberofmaintainers)) people who each merge the change onto their upstream branch as it makes its way to master.